Add some tests for FlagLoader.js

KWierso opened the pull request:
This adds a few tests for functions defined in FlagLoader.js.

There's a test for generating flags for a "firefox39" tree (which doesn't exist, but the flags for it do).

There's a test for generating flags for the "b2g_2_0" tree (which isn't an actual tree, but it's what the flags are called).

There's a test for running the entire FlagLoader.js init function for the "mozilla-b2g32_v2_0" tree (which is what the "b2g_2_0" flags are connected to).

There's a test for running the entire init function for the "mozilla-esr38" tree (which uses the "firefox_esr38" flags).

There's tests for running the entire init function for the "mozilla-central" and "mozilla-aurora" trees (which use version-specific flags based on config files in-tree, requiring network requests to obtain).

KWierso commented:
Arguably those network requests should be mocked out so it just reads static content from the Bugherder repository (especially since the version numbers for their flags will change every time the release cycle moves), but I figure any tests are better than none for the immediate future.

edmorley commented:
hg's `tip` is generally not recommended - 99% of the time `default` is more appropriate.
However in this case I think we should use neither: let's use a specific revision for mozilla-central (eg the revision currently associated with `default`), so the tests don't start failing when the version is bumped on `default`.

KWierso commented:
Okay, added a bunch of b2g tests for the various b2g branches and switched the m-c/m-a tests to fetch specific revisions. All tests passing, so I'm gonna merge this.
